node/core/provisioner/src/lib.rs

```diff
@@ -785,6 +778,39 @@ async fn select_candidates(
 	Ok(candidates)
 }
 
+async fn request_backable_candidates_receipts(
+	selected_candidates: &[(CandidateHash, Hash)],
+	sender: &mut impl overseer::ProvisionerSenderTrait,
+) -> Result<Vec<BackedCandidate>, Error> {
+	let mut response = Vec::with_capacity(selected_candidates.len());
+	let mut selected_iter = selected_candidates.into_iter();
+	let mut candidate = selected_iter.next();
+
+	while let Some((candidate_hash, relay_parent)) = candidate {
+		let mut hashes = vec![*candidate_hash];
+		loop {
+			candidate = selected_iter.next();
+			match candidate {
+				Some((c, rp)) if rp == relay_parent => {
+					hashes.push(*c);
+				},
+				_ => break,
+			};
+		}
+
+		let (tx, rx) = oneshot::channel();
+		sender.send_unbounded_message(CandidateBackingMessage::GetBackedCandidates(
```

Comment:
nit: we could change `GetBackedCandidates` to take a Vec<Hash, Vec<CandidateHash>>> to avoid sending multiple messages to the backing subsystem.